libproto/src/lib.rs

```diff
@@ -234,20 +234,107 @@ impl Block {
         self.get_body().transactions_root().0 == *self.get_header().get_transactions_root()
     }
 
-    pub fn block_verify_req(&self, request_id: u64) -> VerifyBlockReq {
-        let mut reqs: Vec<VerifyTxReq> = Vec::new();
-        let signed_txs = self.get_body().get_transactions();
-        for signed_tx in signed_txs {
-            let signer = signed_tx.get_signer();
-            let unverified_tx = signed_tx.get_transaction_with_sig();
-            let mut verify_tx_req = unverified_tx.tx_verify_req_msg();
-            verify_tx_req.set_signer(signer.to_vec());
-            reqs.push(verify_tx_req);
+    pub fn compact(&self) -> CompactBlock {
+        let mut ret = CompactBlock::new();
+        ret.set_version(self.get_version());
+        ret.set_header(self.get_header().clone());
+        ret.set_body(self.get_body().compact());
+        ret
+    }
+}
+
+impl CompactBlock {
+    pub fn crypt_hash(&self) -> H256 {
+        self.get_header().crypt_hash()
+    }
+
+    pub fn crypt_hash_hex(&self) -> String {
+        self.get_header().crypt_hash_hex()
+    }
+
+    pub fn check_hash(&self) -> bool {
+        self.get_body().transactions_root().0 == *self.get_header().get_transactions_root()
+    }
+
+    pub fn check_txs(&self, stxs: &[SignedTransaction]) -> Result<(), usize> {
+        let hashes = self.get_body().get_tx_hashes();
+        if stxs.len() == hashes.len() {
+            stxs.iter()
+                .zip(hashes.iter())
+                .into_iter()
+                .enumerate()
+                .try_for_each(|(idx, (stx, hash))| {
+                    if &stx.tx_hash == hash {
+                        Ok(())
+                    } else {
+                        Err(idx)
+                    }
+                })
+        } else {
+            Err(0)
```

Comment:
there are two meaning of the `Err(0)`, how to distinguish?

Reply:
How about `Result<(), Enum>`?